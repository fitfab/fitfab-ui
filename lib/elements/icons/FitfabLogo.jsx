import React from 'react'
import { String } from 'prop-types'
import styled from 'styled-components'
import colors from '../../styles/colors'

function Icon({ className, fChar, otherChar, height, width }) {
    console.log(height, width)
    return (
        <svg
            className={className}
            viewBox="0 0 158 45"
            xmlns="http://www.w3.org/2000/svg"
        >
            <g fill="none" fillRule="evenodd" id="Page-1">
                <polygon
                    fill={colors[fChar]}
                    id="f"
                    points="0 0 30 0 30 10 13 10 13 16 27 16 27 26 13 26 13 44 0 44"
                />
                <polygon
                    fill={colors[fChar]}
                    id="f2"
                    points="124 0 158 0 154 10 137 10 137 16 152 16 151 26 137 26 137 44 124 44"
                    transform="matrix(-1 0 0 1 282 0)"
                />
                <polygon
                    fill={colors[otherChar]}
                    id="t"
                    points="30 7.10542736e-15 30 10 44 10 44 44 57 44 57 10 61 10 61 0"
                />
                <polygon
                    fill={colors[otherChar]}
                    id="i"
                    points="27 16 39 16 39 44 27 44"
                />
                <path
                    d="M98,-7.10542736e-15 L98,44.0950248 C79.3575846,43.9981463 69.7041799,43.9981463 69.0397858,44.0950248 C61.0344238,43.2241211 51.1218262,26.5742187 68.0905762,20.0234375 C60.9528809,17.8494466 58.6008301,11.1749674 61.0344238,-7.10542736e-15 L98,-7.10542736e-15 Z M86.1748047,8.00292969 C82.1069337,8.00292969 79.089003,8.00292969 77.1210126,8.00292969 C73.2001953,10.4238281 73.5595703,14.5888672 77.1210126,17.0712891 C78.8544651,17.0712891 81.8723958,17.0712891 86.1748047,17.0712891 L86.1748047,8.00292969 Z M86.1748047,25.0029297 C80.7629991,25.0029297 76.9092738,25.0029297 74.6136287,25.0029297 C70.7507324,27.5683594 71.4306641,32.4628906 74.6136287,34.0712891 C76.9446145,34.0712891 80.7983398,34.0712891 86.1748047,34.0712891 L86.1748047,25.0029297 Z"
                    fill={colors[otherChar]}
                    id="Combined-B"
                />
                <path
                    d="M141.077881,44 L128.07666,44 L125.060547,36.8886719 L110.067139,36.8886719 L108.041504,44 L98,44 L98,33.0515137 L110.393311,0 L124.543701,0 L141.077881,44 Z M118.070557,12.0009766 L117.229492,12.0009766 L112.157227,27.9831543 L123.200684,27.9831543 L118.070557,12.0009766 Z"
                    fill={colors[otherChar]}
                    id="Combined-A"
                />
            </g>
        </svg>
    )
}

Icon.propTypes = {
    className: String,
    fChar: String,
    height: String,
    otherChar: String,
    width: String,
}

Icon.defaultProps = {
    width: '158px',
    height: '48px',
    fChar: 'gray',
    otherChar: 'mediumGray',
}

export const FitfabLogo = styled(Icon)`
    height: ${({ height }) => height};
    width: ${({ width }) => width};
`
