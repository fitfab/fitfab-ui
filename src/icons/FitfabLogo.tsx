import React from 'react'

import styled from 'styled-components'

export interface FitfabProps {
    brandColor?: string
    defaultColor?: string
    width?: string
    height?: string
}
const SVG: React.SFC<FitfabProps> = props => {
    const {
        brandColor = '#FF4158', // default brand color
        defaultColor = '#ADBBC5', // default color
        height = 43,
        width = 159,
        ...rest
    } = props
    return (
        <svg viewBox="0 0 159 43" {...rest}>
            <g fill="none" fillRule="evenodd">
                <path fill={defaultColor} d="M34 .05h39.719v10.606H60.184V43H46.912V10H34z" />
                <path
                    d="M100.486.05H75.643c-4.141 0-7.32 1.026-9.537 3.077-2.216 2.05-3.325 4.59-3.325 7.617 0 2.54.791 4.717 2.373 6.533 1.055 1.211 2.598 2.168 4.63 2.871-3.087.743-5.357 2.017-6.812 3.824-1.455 1.806-2.183 4.077-2.183 6.811 0 2.227.518 4.229 1.553 6.006a11.435 11.435 0 0 0 4.248 4.219c1.113.644 2.793 1.113 5.039 1.406 2.988.39 4.97.586 5.947.586h22.91V.05zM87.098 16.897h-5.772c-2.07 0-3.51-.356-4.321-1.069-.81-.713-1.216-1.743-1.216-3.09 0-1.25.405-2.227 1.216-2.93.81-.703 2.222-1.055 4.233-1.055h5.86v8.144zm0 16.875H80.33c-2.285 0-3.896-.405-4.834-1.215-.937-.81-1.406-1.9-1.406-3.267 0-1.27.464-2.29 1.391-3.061.928-.772 2.554-1.158 4.878-1.158h6.739v8.701zM30.4 17h11.924v26H30.4z"
                    fill={defaultColor}
                />
                <path
                    fill={brandColor}
                    d="M.453.05H34V10H13.783v7h16.64v8.451h-16.64V43H.453zM158.266.05h-32.813V10h19.483v7h-16.641v8.451h16.64V43h13.33z"
                />
                <path
                    d="M126.516 35.91h-15.118l-2.08 7.09H95.754L111.896.05h14.473L142.512 43h-13.887l-2.11-7.09zm-2.784-9.287l-4.746-15.44-4.716 15.44h9.462z"
                    fill={defaultColor}
                />
            </g>
        </svg>
    )
}

export const FitfabLogo = styled(SVG)`
    height: ${p => p.height};
    width: ${p => p.width};
`

FitfabLogo.displayName = 'FitfabLogo'
