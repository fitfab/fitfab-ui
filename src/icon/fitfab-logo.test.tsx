import { cleanup, render } from '@testing-library/react'
import React from 'react'
import { FitfabLogo } from './FitfabLogo'
afterEach(cleanup)

test('Render Fitfab logo with default props', () => {
    const { container } = render(<FitfabLogo />)
    // Snapshot demo
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild!.nodeName).toBe('svg')
})
