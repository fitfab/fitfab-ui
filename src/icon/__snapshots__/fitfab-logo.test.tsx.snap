// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Fitfab logo with default props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    height: ",
          [Function],
          ";
    width: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "FitfabLogo",
      "foldedComponentIds": Array [],
      "render": [Function],
      "self": [Circular],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
